### DISCUSSION TITLE
SRT input to LL-HLS is creating separately named Video & Audio tracks - how can I resolve this using server.xml

### DISCUSSION THREAD

**Discussion by privatesam** [OvenMediaLabs/OvenMediaEngine#1476] [Help]
I'm ingesting a SRT stream into OME in docker.  If I play the published LL-HLS in a player its working fine.  However I'm trying to do something with a third party application which can deal with LL-HLS streams but not if they have separate video and audio tracks (or at least this is what the developer explains).  Indeed when I play my LL-HLS streams from OME in VLC the video track seems to be named Track2 and is not automatically selected (although the video plays):

![Screenshot 2023-12-08 153751](https://github.com/AirenSoft/OvenMediaEngine/assets/15216883/c83eb433-2042-4587-a11a-0ecf8893c909)

and the audio track gets a random name like this:

![audio](https://github.com/AirenSoft/OvenMediaEngine/assets/15216883/8bb3b382-0069-4857-805d-6bf06a36f9d2)

So I suppose my question is two parts:
1.  If it is possible (as the developer I'm liasing with suggests) to create a LL-HLS stream with the audio&video as one track how can I do this with OME?
2. If it's simply that the audio and video tracks are getting different names how can I set the Server.xml to ensure they both get something like "Track 1".

Here is my Server.xml:

`<?xml version="1.0" encoding="UTF-8" ?>

<Server version="8">
	<Name>OvenMediaEngine</Name>
	
	<Type>origin</Type>
	
	<IP>*</IP>
	<PrivacyProtection>false</PrivacyProtection>
	
	<StunServer>stun.ovenmediaengine.com:13478</StunServer>

	<Modules>
		
		<HTTP2>
			<Enable>true</Enable>
		</HTTP2>

		<LLHLS>
			<Enable>true</Enable>
		</LLHLS>

		
		<P2P>
			
			<Enable>false</Enable>
			<MaxClientPeersPerHostPeer>2</MaxClientPeersPerHostPeer>
		</P2P>
	</Modules>

	
	<Bind>
		
		
		<Providers>
			
			<RTSPC>
				<WorkerCount>1</WorkerCount>
			</RTSPC>
			<OVT>
				<WorkerCount>1</WorkerCount>
			</OVT>
			
			<RTMP>
				<Port>${env:OME_RTMP_PROV_PORT:1935}</Port>
				<WorkerCount>1</WorkerCount>
			</RTMP>
			<SRT>
				<Port>${env:OME_SRT_PROV_PORT:9999}</Port>
				<WorkerCount>1</WorkerCount>
			</SRT>
			
			<WebRTC>
				<Signalling>
					<Port>${env:OME_WEBRTC_SIGNALLING_PORT:3333}</Port>
					<TLSPort>${env:OME_WEBRTC_SIGNALLING_TLS_PORT:3334}</TLSPort>
					<WorkerCount>1</WorkerCount>
				</Signalling>
				<IceCandidates>
					<IceCandidate>${env:OME_HOST_IP:*}:${env:OME_WEBRTC_CANDIDATE_PORT:10000-10004/udp}</IceCandidate>
					<TcpRelay>${env:OME_HOST_IP:*}:${env:OME_WEBRTC_TCP_RELAY_PORT:3478}</TcpRelay>
					<TcpForce>true</TcpForce>
					<TcpRelayWorkerCount>1</TcpRelayWorkerCount>
				</IceCandidates>
			</WebRTC>
		</Providers>

		<Publishers>
			
			<OVT>
				<Port>${env:OME_ORIGIN_PORT:9000}</Port>
				<WorkerCount>1</WorkerCount>
			</OVT>
			<LLHLS>
				
				<Port>${env:OME_LLHLS_STREAM_PORT:3333}</Port>
				<TLSPort>${env:OME_LLHLS_STREAM_TLS_PORT:3334}</TLSPort>
				<WorkerCount>1</WorkerCount>
			</LLHLS>
			<WebRTC>
				<Signalling>
					<Port>${env:OME_WEBRTC_SIGNALLING_PORT:3333}</Port>
					<TLSPort>${env:OME_WEBRTC_SIGNALLING_TLS_PORT:3334}</TLSPort>
					<WorkerCount>1</WorkerCount>
				</Signalling>
				<IceCandidates>
					<IceCandidate>${env:OME_WEBRTC_CANDIDATE_IP:*}:${env:OME_WEBRTC_CANDIDATE_PORT:10000-10004/udp}</IceCandidate>
					<TcpRelay>${env:OME_WEBRTC_CANDIDATE_IP:*}:${env:OME_WEBRTC_TCP_RELAY_PORT:3478}</TcpRelay>
					<TcpForce>true</TcpForce>
					<TcpRelayWorkerCount>1</TcpRelayWorkerCount>
				</IceCandidates>
			</WebRTC>
		</Publishers>
	</Bind>

	<VirtualHosts>
		
		<VirtualHost include="VHost*.xml" />
		<VirtualHost>
			<Name>default</Name>
			
			<Distribution>ovenmediaengine.com</Distribution>
			
			
			<Host>
				<Names>
					
					<Name>*</Name>
				</Names>
				<TLS>
					<CertPath>/certs/cert1.pem</CertPath>
					<KeyPath>/certs/privkey1.pem</KeyPath>
					<ChainCertPath>/certs/fullchain1.pem</ChainCertPath>
				</TLS>
			</Host>

			

			

			
			<CrossDomains>
				<Url>*</Url>
			</CrossDomains>

			
			<Applications>
				<Application>
					<Name>app</Name>
					
					<Type>live</Type>
					<OutputProfiles>
						
						

						
						<HWAccels>
							<Decoder>
									<Enable>true</Enable>
									
									
							</Decoder>
							<Encoder>
									<Enable>true</Enable>
									
							</Encoder>
						</HWAccels>

						<OutputProfile>
							<Name>fire</Name>
							<OutputStreamName>${OriginStreamName}</OutputStreamName>

							
							<Encodes>
								<Video>
									<Bypass>true</Bypass>
								</Video>
								<Audio>
									<Bypass>true</Bypass>
								</Audio>
						
								<Audio>
									<Name>aac_audio</Name>
									<Codec>aac</Codec>
									<Bitrate>128000</Bitrate>
									<Samplerate>48000</Samplerate>
									<Channel>2</Channel>
									<BypassIfMatch>
										<Codec>eq</Codec>
									</BypassIfMatch>
								</Audio>								
								<Audio>
									<Name>opus_audio</Name>
									<Codec>opus</Codec>
									<Bitrate>128000</Bitrate>
									<Samplerate>48000</Samplerate>
									<Channel>2</Channel>
									<BypassIfMatch>
										<Codec>eq</Codec>
									</BypassIfMatch>									
								</Audio>
								
								
							</Encodes>
						</OutputProfile>
					</OutputProfiles>
					<Providers>
						<OVT />
						<WebRTC>
							<Timeout>30000</Timeout>
							<CrossDomains>
								<Url>*</Url>
							</CrossDomains>
						</WebRTC>
						<RTMP />
						<SRT />
						<RTSPPull />
						<Schedule>
							<MediaRootDir>/opt/ovenmediaengine/media</MediaRootDir>
							<ScheduleFilesDir>/opt/ovenmediaengine/media</ScheduleFilesDir>
						</Schedule>
						
					</Providers>
					<Publishers>
						<AppWorkerCount>1</AppWorkerCount>
						<StreamWorkerCount>8</StreamWorkerCount>
						<OVT />
						<WebRTC>
							<Timeout>30000</Timeout>
							<Rtx>false</Rtx>
							<Ulpfec>false</Ulpfec>
							<JitterBuffer>false</JitterBuffer>
						</WebRTC>
						<LLHLS>
							<OriginMode>true</OriginMode>
							
							<ChunkDuration>0.5</ChunkDuration>
							
							<PartHoldBack>1.5</PartHoldBack>
							<SegmentDuration>6</SegmentDuration>
							<SegmentCount>10</SegmentCount>
							<CrossDomains>
								<Url>*</Url>
							</CrossDomains>
						</LLHLS>
					</Publishers>
				</Application>
			</Applications>
		</VirtualHost>
	</VirtualHosts>
</Server>`

**Comment by getroot** [comment]
OME cannot combine A/V from LL-HLS into one file. Apple recommends separating A/V into individual files for ABR. What about that player that can't handle A/V separation in LLHLS? Every LL-HLS player I know handles separated A/V well.

Track2 displayed by VLC player does not seem to be provided by OME. And I don't think it will help your problem.

**Comment by privatesam** [comment]
Thanks for the reply.  Fair enough about the separate tracks - you're right of course that all the js players are happy enough but as I mentioned I'm using a 3rd party application which proxies the streams for user management purposes and it can't manage the separate tracks.  I had just wondered if it was a config thing I could change because the docker of the engine has ffmpeg in it and I have the ffmpeg command that can combine and then re-output the stream as one track but obviously this isn't possible.  Thanks for this resource anyway its fantastic!
